BridgeAppSDK/SBAExternalIDOnboardingController.swift

```diff
@@ -53,22 +78,46 @@ public extension SBAExternalIDOnboardingController {
 
     public func registerUser() {
         
-        guard let text: String = self.registrationCodeTextField.text
+        // Check if the text string is valid
+        guard let text: String = self.registrationCodeTextField.text?.trim()
             where text.characters.count > 0 else {
-                let message = NSLocalizedString("Please enter a valid registration code.", comment: "Message for invalid registration code")
+                let message = Localization.localizedString("SBA_REGISTRATION_INVALID_CODE")
                 showAlertWithOk(nil, message: message, actionHandler: nil)
                 return
         }
         
+        // Check if this is a step view controller with verification
+        if let stepVC = self as? SBAExternalIDOnboardingControllerWithVerification,
```

Comment:
don't like an object checking its own type at run-time, maybe put this in a method defined in the base protocol that can be overridden by the derived protocol (and returns true/false so the caller knows whether to exit early)?